src/main/groovy/org/cadixdev/gradle/licenser/LicenseExtension.groovy

```diff
@@ -24,55 +24,56 @@
 
 package org.cadixdev.gradle.licenser
 
+import groovy.transform.PackageScope
 import org.cadixdev.gradle.licenser.header.HeaderFormatRegistry
+import org.gradle.api.Action
 import org.gradle.api.Incubating
 import org.gradle.api.NamedDomainObjectContainer
 import org.gradle.api.Project
-import org.gradle.api.tasks.SourceSet
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.plugins.ExtraPropertiesExtension
+import org.gradle.api.provider.ListProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.provider.ProviderFactory
+import org.gradle.api.resources.TextResourceFactory
 import org.gradle.api.tasks.util.PatternFilterable
 import org.gradle.api.tasks.util.PatternSet
 import org.gradle.util.ConfigureUtil
 
+import javax.inject.Inject
+
 /**
  * Represents the Gradle extension for configuring the settings for the
  * {@link Licenser} plugin.
  */
 class LicenseExtension extends LicenseProperties {
 
-    /**
-     * The source sets to scan for files with license headers.
-     * By default this includes all source sets of the project.
-     */
-    Collection<SourceSet> sourceSets
-
-    /**
-     * The Android source sets to scan for files with license headers.
-     * By default this includes all source sets of the project.
-     *
-     * <p><b>Note:</b> This is only used when the Android Gradle plugin is applied.</p>
-     */
-    @Incubating
-    Collection androidSourceSets
-
     /**
      * The charset to read/write the files with.
      * By default this is {@code UTF-8}.
      */
-    String charset = 'UTF-8'
+    final Property<String> charset
 
     /**
      * Whether to ignore failures and only warn about license violations
      * instead of failing the build.
      * By default this is {@code false}.
      */
-    boolean ignoreFailures = false
+    final Property<Boolean> ignoreFailures
 
     /**
      * Whether to skip existing license headers instead of failing the build or
      * updating the license headers.
      * By default this is {@code false}.
      */
-    boolean skipExistingHeaders = false
+    final Property<Boolean> skipExistingHeaders
+
+    /**
+     * The line ending to use within license headers.
+     *
+     * By default this is {@link System#lineSeparator()}
```

Comment:
Do we want this as the default? cc @Minecrell

Reply:
Ideally it would try to detect which line ending the file is using. This is because it might vary from file to file. For example, in licenser's `.gitattributes` I configure it as follows:

https://github.com/CadixDev/licenser/blob/53e0f575f89a0225be02da0d271aaf56a54e1c06/.gitattributes#L1-L4

Most files use system line endings, but `*.bat` is always explicitly checked out as CRLF, and `gradlew`, `*.sh` as LF. I don't remember exactly why I set it up like this, perhaps it's from the time when I was still using Windoofs and I was trying to run some bash script there. Or it's because `gradle wrapper` generates it like this and I got warnings from git after regenerating the wrapper.

Reply:
But given that the default so far was system line endings (AFAICT), I suppose this is fine for now. Question is just if we end up deprecating this option fairly quickly if someone decides to work on the automatic detection of line endings.

Reply:
I had a quick chat with jamie on IRC -- ideally you would be able to auto-detect from files, but that takes some actual thought to implement so I haven't tackled that in this PR. When that is implemented, I was thinking auto-detection could be represented as a `null` value for this property.